miasm2/core/graph.py

```diff
@@ -149,3 +159,44 @@ def dot(self):
                                                 self.edge2str(a, b))
         out += "}"
         return out
+
+
+
+    def compute_dominators(self):
+        """Compute dominators of the graph"""
+        nodes = self.nodes()
+        heads = self.heads()
```

Comment:
You have add the `heads_iter` methods, which returns an iterator. If you use `heads` method instead, the iterator will be consumed *before* returning to `compute_dominators`. That way, the iterator's interest is lost.

I think you can safely replace `heads` with `heads_iter` here.

`Difference` still working:
``` Python
>>> set("test").difference(xrange(3))
set(['s', 'e', 't'])
```